modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PartitionMetaFactory.java

```diff
@@ -25,7 +25,20 @@
  * Factory for creating {@link PartitionMeta} instances.
  */
 public interface PartitionMetaFactory {
+    /**
+     * Creates a new {@link PartitionMeta} instance.
+     *
+     * @param checkpointId Checkpoint ID.
+     * @param metaIo Partition meta IO.
+     * @param pageAddr Page address.
+     * @return Partition meta.
+     */
     PartitionMeta createPartitionMeta(@Nullable UUID checkpointId, PartitionMetaIo metaIo, long pageAddr);
 
+    /**
+     * Creates a new {@link PartitionMetaIo} instance.
```

Comment:
I don't think it create a new instance; it rather returns an IO of the 'current' version.

Reply:
done